### PR TITLE
feat(events): Add remaining Apple account security events

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We need to support security events from Apple

## This pull request

- Adds support for the email disabled, email enabled, consent-revoked and account delete events

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-7978
Closes: https://mozilla-hub.atlassian.net/browse/FXA-8001
Closes: https://mozilla-hub.atlassian.net/browse/FXA-8002

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/utils/third-party-events.ts [hunk 1]
```diff
@@ -57,33 +75,84 @@ async function revokeThirdPartySessions(
...
-    handleAccountDisabledEvent,
+    handleGoogleAccountDisabledEvent,
 
   'https://schemas.openid.net/secevent/risc/event-type/account-credential-change-required':
-    handleSessionsRevokedEvent,
+    handleGoogleSessionsRevokedEvent,
+};
+
+// See https://developer.apple.com/documentation/sign_in_with_apple/processing_changes_for_sign_in_with_apple_accounts
+export const appleEventHandlers = {
```
Comment (vbudhram) [inline packages/fxa-auth-server/lib/routes/utils/third-party-events.ts:105]: It turns out that Apple does not conform to the RISC standard so I had to modify the code a bit to support this.  I probably over engineered it in the original Google PR, but maybe that is ok.

File: packages/fxa-auth-server/lib/routes/linked-accounts.ts [hunk 2]
```diff
@@ -496,6 +545,19 @@ export const linkedAccountRoutes = (
       },
       handler: async (request: AuthRequest) => handler.handleGoogleSET(request),
     },
+    {
+      method: 'POST',
+      path: '/linked_account/webhook/apple_event_receiver',
+      options: {
+        payload: {
+          // Security events use the content type application/secevent+jwt,
+          // It isn't clearly documented, but the payload is a JWT buffer.
```
Comment (chenba) [inline packages/fxa-auth-server/lib/routes/linked-accounts.ts:554]: 👍

File: packages/fxa-auth-server/lib/routes/utils/third-party-events.ts [hunk 3]
```diff
@@ -57,33 +75,84 @@ async function revokeThirdPartySessions(
...
 
-function handleTestEvent(eventDetails: SETEvent, log: any) {
+/**
+ * Handle the consent revoked event. Apple recommends that we revoke all sessions
+ * and unlink the account.
+ *
+ * @param eventDetails
+ * @param log
+ * @param db
+ * @returns Promise<void>)
```
Comment (chenba) [inline packages/fxa-auth-server/lib/routes/utils/third-party-events.ts:121]: Got an errant `)` there.

**Review by chenba — APPROVED** [review]
LGTM

**Comment by vbudhram** [comment]
@chenba Thank you!

**Pull request merged**